### ISSUE TITLE
fetch_trades mem usage

### ISSUE THREAD

**Issue by elabunsky** [ccxt/ccxt#18662]
### Operating System

ubuntu

### CCXT Version

last

### Description

Hi,

Is it possible control mem usage for fetch_trades? 

![image](https://github.com/ccxt/ccxt/assets/33313598/41bcc841-b54c-405e-a52d-597301588e30)

Regards,
Eugene.

### Code

```
  

```

**Comment by mrpouley** [comment]
Hi,

I can confirm that memory usage slowly increases over time and never seems to reach a limit.

ccxt 4.0.27

```
import ccxt.pro as ccxt
import asyncio
import resource
import time


bybit = ccxt.bybit({"tradesLimit": 10})

start = time.time()


async def watch_symbol(symbol) -> None:
    while True:
        await bybit.watch_trades(symbol=symbol)

        print(
            "Time:", int(time.time() - start), "s, ",
            "RAM:", resource.getrusage(resource.RUSAGE_SELF).ru_maxrss  # kilobytes
        )


async def main() -> None:
    symbols = (await bybit.load_markets()).keys()

    await asyncio.gather(*[
        watch_symbol(symbol) for symbol in symbols if symbol.endswith("USDT")
    ])


asyncio.run(main())
```

```
...
Time: 62414 s,  RAM: 794804
```

**Comment by ttodua** [comment]
old topic, but just for reference /cc @pcriadoperez

**Comment by adahan** [comment]
I confirm I observe a similar effect grows to 4.8 GB in 33 hours at 31 KB/sec will try to use jemalloc see if it solves the issue

**Comment by carlotestor** [comment]
Yes — memory is controlled by `tradesLimit`, which caps the per-symbol ring buffer of cached trades (default **1000** per symbol):

```python
exchange = ccxt.pro.bybit({'options': {'tradesLimit': 100}})
```

Note the option belongs under `options`, not at the top level — in @mrpouley's repro `{"tradesLimit": 10}` was passed as a *constructor* property, so it was ignored and the default 1000 stayed in effect. That alone explains most of the observed growth: ~500 USDT symbols × 1000 trades each is genuinely hundreds of MB of Python objects.

Other knobs of the same kind, if you watch other streams:
```python
'options': {
    'tradesLimit': 100,
    'ordersLimit': 100,
    'myTradesLimit': 100,
    'OHLCVLimit': 100,
}
```

Also worth knowing:
- The caches are bounded, so RSS should plateau — but CPython rarely returns freed memory to the OS, so `ru_maxrss` keeps looking like a leak even after it stops growing. @adahan's jemalloc idea addresses exactly this (arena fragmentation), and usually helps.
- Watching every symbol on one connection is expensive by nature; subscribing only to the symbols you actually trade is the biggest single win.

If setting `tradesLimit` under `options` flattens the curve for you, this can be closed.
